File: src/main/java/com/pythonbyte/spring_into_ai/entity/ChatAuditLog.java
package com.pythonbyte.spring_into_ai.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "chat_audit_logs")
public class ChatAuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String prompt;

    @Column(nullable = false)
    private String response;

    @Column(name = "created_date", nullable = false)
    private LocalDateTime createdDate;

    public ChatAuditLog() {
        this.createdDate = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getPrompt() {
        return prompt;
    }

    public void setPrompt(String prompt) {
        this.prompt = prompt;
    }

    public String getResponse() {
        return response;
    }

    public void setResponse(String response) {
        this.response = response;
    }

    public LocalDateTime getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(LocalDateTime createdDate) {
        this.createdDate = createdDate;
    }
} 

File: src/main/kotlin/com/pythonbyte/spring_into_ai/entities/ChatAuditLog.kt
package com.pythonbyte.spring_into_ai.entities

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "chat_audit_logs")
data class ChatAuditLog(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    var prompt: String = "",

    @Column(nullable = false)
    var response: String = "",

    @Column(name = "created_date", nullable = false)
    val createdDate: LocalDateTime = LocalDateTime.now()
)